core/src/main/java/org/fao/geonet/api/records/attachments/StoreUtils.java

```diff
@@ -80,6 +77,82 @@ public static void copyDataDir(ServiceContext context, String oldUuid, String ne
         }
     }
 
+    /**
+     * Ues to to remove items from a metadataResource list
+     *
+     * @param l left list
+     * @param r right list
+     * @return l except r
+     */
+    private static List<MetadataResource> exceptMetadataResource(List<MetadataResource> l, List<MetadataResource> r)  {
+        Map<String,MetadataResource> rMap = new HashMap<>();
+        for (MetadataResource mr : r) rMap.put(mr.getId(),mr);
+
+        Iterator<MetadataResource> itr = l.iterator();
```

Comment:
This code is updating the list in the parameter `l`, can cause some side effects. I would create a copied list instead instead of relying to clone the list in the function call as done later:

```
 exceptMetadataResource(new ArrayList<>(targetResources), sourceResources);
```